### DISCUSSION TITLE
How do you compress a file via ZIP or via LZMA?

### DISCUSSION THREAD

**Discussion by houmie** [tsolomko/SWCompression#48] [Q&A] [answered]
Hello,

I came across this project today and it seems great as it supports multiple compression formats.
However the main page on Github has very limited examples (only unarchive and decompress), but there isn't a single compression example.

Is there any example you could provide how I could compress all files in a directory via Zip or alternatively via LZMA respectively?

This is how I do it with SSArchive right now. logFilePaths is the path of my daily log files, which I'd compress and attach to the email

```
if attachLog {
     let logFilePaths = LogService.shared.fileLogger!.logFileManager.sortedLogFilePaths
        if let zipFilePath = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first {
            let zipFile = zipFilePath.appendingPathComponent("logs.zip")
            SSZipArchive.createZipFile(atPath: zipFile.path, withFilesAtPaths: logFilePaths)
            if let fileData = NSData(contentsOfFile: zipFile.path) {
            mail.addAttachmentData(fileData as Data, mimeType: "application/zip", fileName: "logs.zip")
        }
    }
}
```

Can SWCompression do this as well?

Many Thanks

**Comment by tsolomko** [comment, marked as answer]
Hello,
 
> Can SWCompression do this as well?

Unfortunately, the answer to this question is no, since neither LZMA compression nor creation of ZIP archives is implemented. Furthermore, even if they were implemented, SWCompression by design (i.e. intentionally) would not interact with the file system directly (as seemingly SSArchive does in your example), so you would have to implement this part of the procedure by yourself.

> Is there any example you could provide how I could compress all files in a directory via Zip or alternatively via LZMA respectively?

Some examples of how to implement the interactions with the file system (for the functionality that is currently available in SWCompression) may be found in the ["sophisticated example"](https://github.com/tsolomko/SWCompression#sophisticated-example) mentioned in README.
